index.js

```diff
@@ -0,0 +1,408 @@
+  /**
+   * Custom helper functions
+   */
+  onShowModal: function (event) {
+    this.connector.enable();
+  },
+  onCloseModal: function (event) {
+    this.el.emit("avatar-connect:close-modal");
+  },
+  createAvatar: function (avatar) {
+    var avatarEl = document.createElement("a-entity");
+    // TODO: should be on and off turnable
+    avatarEl.setAttribute("avatar-connect-avatar", {
+      uri: avatar.uri,
+      format: avatar.format,
+      showPodest: true,
+    });
+    this.el.appendChild(avatarEl);
+  },
+});
+
+/**
+ * AvatarConnect Avatar component for A-Frame.
+ */
+AFRAME.registerComponent("avatar-connect-avatar", {
+  schema: {
+    uri: { default: "" },
+    format: { default: "glb" },
+    showPodest: { default: true },
+  },
+
+  /**
+   * Set if component needs multiple instancing.
+   */
+  multiple: false,
+
+  /**
+   * Called once when component is attached. Generally for initial setup.
+   */
+  init: async function () {
+    if (this.data.uri) {
+      await this.createAvatar();
+    } else {
+      this.el.emit("avatar-connect:error", "Avatar URI is empty");
+      console.error("URI is empty");
+    }
+  },
+
+  /**
+   * Called when component is attached and when component data changes.
+   * Generally modifies the entity based on the data.
+   */
+  update: function (oldData) {},
+
+  /**
+   * Called when a component is removed (e.g., via removeAttribute).
+   * Generally undoes all modifications to the entity.
+   */
+  remove: function () {},
+  /**
+   * Called on each scene tick.
+   */
+  // tick: function (t) { },
+
+  /**
+   * Called when entity pauses.
+   * Use to stop or remove any dynamic or background behavior such as events.
+   */
+  pause: function () {},
+
+  /**
+   * Called when entity resumes.
+   * Use to continue or add any dynamic or background behavior such as events.
+   */
+  play: function () {},
+
+  /**
+   * Event handlers that automatically get attached or detached based on scene state.
+   */
+  events: {
+    // click: function (evt) { }
+  },
+
+  /**
+   * Custom helper functions
+   */
+  createAvatar: async function () {
+    var avatarEl = document.createElement("a-entity");
+    // TODO: support more formats and providers
+    switch (this.data.format) {
+      case "glb":
+        const gender = await this.getRPMGender(this.data.uri);
+        avatarEl.setAttribute("gltf-model", this.data.uri);
+        // This is experimental. Adds a rig animation to RPM avatars
+        avatarEl.setAttribute("rig-animation", {
+          remoteId: gender,
+          clip: "IDLE",
+          loop: "repeat",
+          crossFadeDuration: 0,
+        });
+        break;
+      case "vrm":
+        // As VRM is based on GLTF, we can load it with the same loader
+        avatarEl.setAttribute("gltf-model", this.data.uri);
+        // VRM specification requires humanoid models to look into -z direction
+        // avatarEl.setAttribute("rotation", "0 -180 0");
+        break;
+      default:
```

Comment:
We should consider how this might scale with infinite numbers of avatar projects. Ideally, we're relying exclusively on the metadata from the bridge response and handling this logic using a provider-agnostic approach.

Using a switch-style approach for each format is the right call here, but if a specific provider needs additional modifications (i.e. RPM with gender), we might want to allow them to add their own customizable submodule (either to this package itself or as an plugin-style package an experience would install separately if they're using that provider).

We'll likely get a better idea of how this looks and works best as we get more and more providers, but off the top of my head, we might want a map of "provider loader modules" that looks like this:

```javascript
const PROVIDER_LOADERS = {
   'ready-player-me': function(avatarEl) {
      avatarEl.setAttribute("rig-animation", {
          remoteId: this.data.metadata.outfitGender, // See my next comment about this
          clip: "IDLE",
          loop: "repeat",
          crossFadeDuration: 0,
      )
   }
}
```

Then, in the switch statement, we'd just do something like this:

```javascript
var avatarEl = document.createElement('a-entity')

switch (this.data.avatar.format) {
  case 'glb':
  case 'vrm':
    avatarEl.setAttribute('gltf-model', this.data.avatar.uri)
    break
  default:
    return
}

const providerLoader = PROVIDER_LOADERS[this.data.providerId]

// If the provider has customized their loader, handle it separately
if (providerLoader) await providerLoader.call(avatarEl, this.data) // use `call` here so the module has access to the same `this` context
```